js/main.js

```diff
@@ -28,21 +28,32 @@ export function isDarkMode() {
     return window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;
 }
 
-compiler.setLogCallback(function (str) {
+function escapeHTML(str) {
+    var escapedHTML = document.createElement("div");
+    escapedHTML.innerText = str;
+    return escapedHTML.innerHTML;
+}
+
+compiler.setLogCallback(function (str, kind) {
     var output = document.getElementById("output");
-    if (str === null) {
-        output.value = "";
+    if (str === null && kind == null) {
```

Comment:
```suggestion
    if (str === null && kind === null) {
```

Reply:
Actually the other way around, nowhere else in this codebase uses `===`. (Cleaning it up to `"use strict"` can be a separate task!) (And don't worry, `"" != null` and `0 != null` and `null == null`, 90s JS wasn't *that* crazy.)

Reply:
Yeah, it'd be nice to use it everywhere else, but I see no harm in keeping the expression self-consistent. (You're more knowledgeable about web dev than I, so you'd know better how to set up some linting or whatever... Feel free to open other PRs if you'd like!)